code/controllers/subsystem/ships.dm

```diff
@@ -132,10 +132,17 @@ SUBSYSTEM_DEF(ship)
 			C = pick(S.ship_components)
 			if(C.active) C = null
 		if(C)
+			C.health += 1 //Only have time for basic functionality.
 			C.active = 1 //fix that shit
 
 			broadcast_message("<span class=notice>[faction2prefix(S)] ship ([S.name]) has repaired [C.name] at ([C.x_loc],[C.y_loc]).</span>",notice_sound,S)
 
+/datum/controller/subsystem/ship/proc/restore_component_power(var/datum/ship_component/C)
+	if(C) //haha runtimes am I right?
+		if(C.ship) //TODO, fix the need for these checks
```

Comment:
todo: kill the fucking ship subsystem. I did a large rework of it once but never finished it so I can try add that once we rebase.

also theres comment syntax for properly showing todo tasks. I think its TODO: